spark/src/main/scala/org/apache/iceberg/spark/SparkTableUtil.scala

```diff
@@ -131,18 +136,22 @@ object SparkTableUtil {
         s"$name=${partition(name)}"
       }.mkString("/")
 
-      DataFiles.builder(spec)
-          .withPath(path)
-          .withFormat(format)
-          .withPartitionPath(partitionKey)
-          .withFileSizeInBytes(fileSize)
-          .withMetrics(new Metrics(rowCount,
-            arrayToMap(columnSizes),
-            arrayToMap(valueCounts),
-            arrayToMap(nullValueCounts),
-            arrayToMap(lowerBounds),
-            arrayToMap(upperBounds)))
-          .build()
+      var builder = DataFiles.builder(spec)
+        .withPath(path)
+        .withFormat(format)
+        .withFileSizeInBytes(fileSize)
+        .withMetrics(new Metrics(rowCount,
+          arrayToMap(columnSizes),
+          arrayToMap(valueCounts),
+          arrayToMap(nullValueCounts),
+          arrayToMap(lowerBounds),
+          arrayToMap(upperBounds)))
+
+      if (partitionKey == "") {
+        builder.build()
+      } else {
+        builder.withPartitionPath(partitionKey).build()
```

Comment:
Not directly related to PR: it seems we don't support dates in `withPartitionPath` as `fromPartitionString` in `Conversions` doesn't handle dates. @chenjunjiedada @rdblue am I correct?

Reply:
Should we have something like below?

```
case DATE:
  return Literal.of(asString).to(Types.DateType.get()).value();
```

Reply:
Let me create an issue for this. We can close it later on if we don't want to fix it.